src/globus_sdk/globus_app/app.py

```diff
@@ -378,13 +378,14 @@ def _auth_params_with_required_scopes(
 
         if not auth_params:
             auth_params = GlobusAuthorizationParameters()
+        parsed_required_scopes = []
+        for s in auth_params.required_scopes or []:
+            parsed_required_scopes.extend(Scope.parse(s))
 
         # merge scopes for deduplication to minimize url request length
         # this is useful even if there weren't any auth_param scope requirements
         # as the app's scope_requirements can have duplicates
-        combined_scopes = ScopeParser.merge_scopes(
-            required_scopes, [Scope(s) for s in auth_params.required_scopes or []]
-        )
+        combined_scopes = Scope.merge_scopes(required_scopes, parsed_required_scopes)
```

Comment:
We need to resolve these changes into the 4.x style.

- `ScopeParser.parse` should be used above, not `Scope.parse`
- `ScopeParser.merge_scopes` is now the call